libqtile/layout/tile.py

```diff
@@ -115,9 +115,16 @@ def down(self):
     def reset_master(self, match=None):
         if not match and not self.master_match:
             return
-        match = match or self.master_match
+        master_match = match or self.master_match
         if self.clients:
-            masters = [c for c in self.clients if match.compare(c)]
+            if type(master_match) is list:
+                masters = []
+                for c in self.clients:
+                    for match in master_match:
+                        if match.compare(c):
+                            masters.append(c)
+            else:
+                masters = [c for c in self.clients if master_match.compare(c)]
```

Comment:
Perhaps if we make `self.master_match` a list if it isn't already in `__init__()`, then we can do something like this:
```suggestion
        if self.clients:
	    master_match = match or self.master_match
            if isinstance(master_match, Match):
		mastermatch = [mastermatch]
	    masters = []
	    for c in self.clients:
		for match in master_match:
		    if match.compare(c):
			masters.append(c)
```

Which i think is a bit easier to read